### PR TITLE
fix!: columns table in information_schema misses some columns

### LINKED ISSUE #3606: The columns table in information_schema is not compatible with MySQL
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Table Engine

### Minimal reproduce step

Missing lots of columns:

```sql
mysql> DESC COLUMNS;
+----------------+--------+------+------+---------+---------------+
| Column         | Type   | Key  | Null | Default | Semantic Type |
+----------------+--------+------+------+---------+---------------+
| table_catalog  | String |      | NO   |         | FIELD         |
| table_schema   | String |      | NO   |         | FIELD         |
| table_name     | String |      | NO   |         | FIELD         |
| column_name    | String |      | NO   |         | FIELD         |
| data_type      | String |      | NO   |         | FIELD         |
| semantic_type  | String |      | NO   |         | FIELD         |
| column_default | String |      | YES  |         | FIELD         |
| is_nullable    | String |      | NO   |         | FIELD         |
| column_type    | String |      | NO   |         | FIELD         |
| column_comment | String |      | YES  |         | FIELD         |
+----------------+--------+------+------+---------+---------------+
```

But MySQL `columns` table has many other columns:

https://dev.mysql.com/doc/refman/8.0/en/information-schema-columns-table.html

### What did you expect to see?

As described above.

### What did you see instead?

As described above.

### What operating system did you use?

All os

### What version of GreptimeDB did you use?

0.7.1

### Relevant log output and stack trace

_No response_

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## Refer to a related PR or issue link (optional)

Fix #3606 

## What's changed and what's your intention?

* Fixed `information_schema.columns`  missing some columns. #3606
* Use sql data type as `data_type` in `information_schema.columns`.
* Refactor `show columns` statement execution.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [ ]  This PR does not require documentation updates.

### CODE REVIEW

File: src/catalog/src/information_schema/columns.rs [hunk 1]
```diff
@@ -236,7 +332,94 @@ impl InformationSchemaColumnsBuilder {
...
+                    self.numeric_scales.push(None);
+                }
+                ConcreteDataType::Float64(_) => {
+                    self.numeric_precisions.push(Some(22));
+                    self.numeric_scales.push(None);
+                }
+                ConcreteDataType::Decimal128(decimal_type) => {
+                    self.numeric_precisions
+                        .push(Some(decimal_type.precision() as i64));
+                    self.numeric_scales.push(Some(decimal_type.scale() as i64));
```
Comment (sunng87) [inline src/catalog/src/information_schema/columns.rs:382]: how about moving `numeric_precisions` and `numeric_scales` to `ConcreteDataType`?

**Review by sunng87 — COMMENTED** [review]
Do we need full compatibility with MySQL? `SRS_ID` is still missing although it's all NULL for us.

File: src/catalog/src/information_schema/columns.rs (hunk 1 above)
Comment (killme2008) [reply src/catalog/src/information_schema/columns.rs:382]: Looks good to me, I'll change it.

**Comment by killme2008** [comment]
> Do we need full compatibility with MySQL? `SRS_ID` is still missing although it's all NULL for us.

Yes, forgot this column.

**Comment by killme2008** [comment]
@fengjiachun @waynexia Please take a look

File: src/datatypes/src/data_type.rs [hunk 2]
```diff
@@ -266,6 +266,38 @@ impl ConcreteDataType {
...
+            ConcreteDataType::Int64(_) => Some(19),
+            ConcreteDataType::UInt64(_) => Some(20),
+            ConcreteDataType::Float32(_) => Some(12),
+            ConcreteDataType::Float64(_) => Some(22),
+            ConcreteDataType::Decimal128(decimal_type) => Some(decimal_type.precision()),
+            _ => None,
+        }
+    }
+
+    /// Try to get numeric scale, returns `None` if it's not numeric type
```
Comment (fengjiachun) [inline src/datatypes/src/data_type.rs:284]:
```suggestion
    /// Try to get numeric scale, returns `None` if it's float or not numeric type
```

File: src/query/src/sql.rs [hunk 3]
```diff
@@ -65,6 +65,7 @@ const TABLES_COLUMN: &str = "Tables";
 const FIELD_COLUMN: &str = "Field";
 const TABLE_TYPE_COLUMN: &str = "Table_type";
 const COLUMN_NAME_COLUMN: &str = "Column";
+const COLUMN_GREPTIME_TYPE_COLUMN: &str = "Greptime_Type";
```
Comment (fengjiachun) [inline src/query/src/sql.rs:68]: `Greptime_Type` or `Greptime_type`? Because I saw `Table_type` at line 66

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/query/src/sql.rs (hunk 3 above)
Comment (killme2008) [reply src/query/src/sql.rs:68]: Make sense, i'll change it.

**Comment by killme2008** [comment]
@sunng87 PTAL

**Pull request merged**